### PR TITLE
Code Review Commit

### PR DESCRIPTION
**Pull request by Mohini-Chakravarty**
Changes Made -

1. Fixed Empty Search Error
2. Implemented variable declaration for ingredients (previously hardcoded to 2g pizza)
3. Made Header Icons(Moms Spaghetti, Home, Privacy) more readable by increasing the font size

Learnings -

1. It is important to handle exception in codes.
2. It is not recommended to use hardcoded values
3. Readability in UI is crucial to bring about a good a web page

Was the project available on time?
Yes, it was available on time

Does the project compile ?
Yes, it compile without throwing any errors.

Does the project have adequate documentation ?
The code has been laid out with suitable comments wherever necessary making it very easy to understand.

### CODE REVIEW

File: MomsSpaghetti/Pages/Index.cshtml [hunk 1]
```diff
@@ -225,15 +225,31 @@
...
+                                    @*<li class="recipe__item">
+                    <svg class="recipe__icon">
+                        <use href="img/icons.svg#icon-check"></use>
+                    </svg>
+                    <div class="recipe__count">2</div>
+                    <div class="recipe__ingredient">
+                        <span class="recipe__unit">g</span>
+                        pizza
+                    </div>
+                </li>*@
```
Comment (Mohini-Chakravarty) [inline MomsSpaghetti/Pages/Index.cshtml:237]: For every ingredient in the recipe, this code will loop through the recipes and map the ingredients using @ingredient.ToString() in the index.cshtml file. It was initially hardcoded with 2, g and pizza.

File: MomsSpaghetti/Pages/Index.cshtml [hunk 2]
```diff
@@ -225,15 +225,31 @@
...
+                                            <p>@ingredient.ToString()</p>
+
                                         </div>
+                                        @*
+                      <div class="recipe__count">2</div>
+                    <div class="recipe__ingredient">
+                        <span class="recipe__unit">g</span>
+                        pizza
+                    </div>
+                                        *@
```
Comment (Mohini-Chakravarty) [inline MomsSpaghetti/Pages/Index.cshtml:252]: For every ingredient in the recipe, this code will loop through the recipes and map the ingredients using @ingredient.ToString() in the index.cshtml file. It was initially hardcoded with 2, g and pizza.

File: MomsSpaghetti/Pages/Index.cshtml [hunk 3]
```diff
@@ -537,6 +553,19 @@
...
+                            if ($(".search__field").val() == "") {
+
+                                alert("Enter a value in the Search Field");
+
+
+                                event.preventDefault();
+
+
+                            }
+                        });
```
Comment (Mohini-Chakravarty) [inline MomsSpaghetti/Pages/Index.cshtml:568]: This validation code will ensure that no empty values are entered by the user in the search field. Using the onclick function, the code checks if the search field value is empty. If empty, it alerts the user to enter a value in the search field.

File: MomsSpaghetti/Pages/Properties/launchSettings.json [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+      "environmentVariables": {
+        "ASPNETCORE_ENVIRONMENT": "Development"
+      }
+    },
+    "MomsSpaghetti": {
+      "commandName": "Project",
+      "launchBrowser": true,
+      "applicationUrl": "https://localhost:5001;http://localhost:5000",
+      "environmentVariables": {
+        "ASPNETCORE_ENVIRONMENT": "Development"
```
Comment (Mohini-Chakravarty) [inline MomsSpaghetti/Pages/Properties/launchSettings.json:23]: No changes made

File: MomsSpaghetti/Pages/Shared/_Layout.cshtml [hunk 5]
```diff
@@ -13,18 +13,18 @@
     <header>
         <nav class="navbar navbar-expand-sm navbar-toggleable-sm navbar-light bg-white border-bottom box-shadow mb-3">
             <div class="container">
-                <a class="navbar-brand" asp-area="" asp-page="/Index">MomsSpaghetti</a>
+                <a class="navbar-brand" asp-area="" asp-page="/Index"><h3>MomsSpaghetti</h3></a>
```
Comment (Mohini-Chakravarty) [inline MomsSpaghetti/Pages/Shared/_Layout.cshtml:16]: Increased font size

File: MomsSpaghetti/Pages/Shared/_Layout.cshtml [hunk 6]
```diff
@@ -13,18 +13,18 @@
...
                 <button class="navbar-toggler" type="button" data-toggle="collapse" data-target=".navbar-collapse" aria-controls="navbarSupportedContent"
                         aria-expanded="false" aria-label="Toggle navigation">
                     <span class="navbar-toggler-icon"></span>
                 </button>
-                <div class="navbar-collapse collapse d-sm-inline-flex flex-sm-row-reverse">
+                <div class="navbar-collapse collapse d-sm-inline-flex flex-sm-row-reverse ">
                     <ul class="navbar-nav flex-grow-1">
                         <li class="nav-item">
-                            <a class="nav-link text-dark" asp-area="" asp-page="/Index">Home</a>
+                            <a class="nav-link text-dark text" asp-area="" asp-page="/Index"><h3>Home</h3></a>
```
Comment (Mohini-Chakravarty) [inline MomsSpaghetti/Pages/Shared/_Layout.cshtml:24]: Increased font size

File: MomsSpaghetti/Pages/Shared/_Layout.cshtml [hunk 7]
```diff
@@ -13,18 +13,18 @@
...
                         </li>
                         <li class="nav-item">
-                            <a class="nav-link text-dark" asp-area="" asp-page="/Privacy">Privacy</a>
+                            <a class="nav-link text-dark" asp-area="" asp-page="/Privacy"><h3>Privacy</h3></a>
```
Comment (Mohini-Chakravarty) [inline MomsSpaghetti/Pages/Shared/_Layout.cshtml:27]: Increased font size

File: MomsSpaghetti/Pages/Shared/_Layout.cshtml [hunk 8]
```diff
@@ -13,18 +13,18 @@
...
         <nav class="navbar navbar-expand-sm navbar-toggleable-sm navbar-light bg-white border-bottom box-shadow mb-3">
             <div class="container">
-                <a class="navbar-brand" asp-area="" asp-page="/Index">MomsSpaghetti</a>
+                <a class="navbar-brand" asp-area="" asp-page="/Index"><h3>MomsSpaghetti</h3></a>
                 <button class="navbar-toggler" type="button" data-toggle="collapse" data-target=".navbar-collapse" aria-controls="navbarSupportedContent"
                         aria-expanded="false" aria-label="Toggle navigation">
                     <span class="navbar-toggler-icon"></span>
                 </button>
-                <div class="navbar-collapse collapse d-sm-inline-flex flex-sm-row-reverse">
+                <div class="navbar-collapse collapse d-sm-inline-flex flex-sm-row-reverse ">
```
Comment (Mohini-Chakravarty) [inline MomsSpaghetti/Pages/Shared/_Layout.cshtml:21]: No changes made

File: MomsSpaghetti/Pages/Index.cshtml [hunk 9]
```diff
@@ -225,15 +225,31 @@
 
                                 @foreach (var ingredient in Model.Recipe.Ingredients)
                                 {
+                                    @*<li class="recipe__item">
+                    <svg class="recipe__icon">
+                        <use href="img/icons.svg#icon-check"></use>
+                    </svg>
+                    <div class="recipe__count">2</div>
+                    <div class="recipe__ingredient">
+                        <span class="recipe__unit">g</span>
```
Comment (discospiff) [inline MomsSpaghetti/Pages/Index.cshtml:234]: What's g?

File: MomsSpaghetti/Pages/Index.cshtml [hunk 10]
```diff
@@ -225,15 +225,31 @@
 
                                 @foreach (var ingredient in Model.Recipe.Ingredients)
                                 {
+                                    @*<li class="recipe__item">
+                    <svg class="recipe__icon">
+                        <use href="img/icons.svg#icon-check"></use>
+                    </svg>
```
Comment (discospiff) [inline MomsSpaghetti/Pages/Index.cshtml:231]: What's svg?

File: MomsSpaghetti/Pages/Index.cshtml [hunk 11]
```diff
@@ -225,15 +225,31 @@
...
+                    <div class="recipe__count">2</div>
+                    <div class="recipe__ingredient">
+                        <span class="recipe__unit">g</span>
+                        pizza
```
Comment (discospiff) [inline MomsSpaghetti/Pages/Index.cshtml:235]: pizza?

File: MomsSpaghetti/Pages/Index.cshtml (hunk 1 above)
Comment (discospiff) [reply MomsSpaghetti/Pages/Index.cshtml:237]: I'll confess I don't entirely understand this change, as I still see 2, g, and pizza.

File: MomsSpaghetti/Pages/Index.cshtml (hunk 3 above)
Comment (discospiff) [reply MomsSpaghetti/Pages/Index.cshtml:568]: Good.

File: MomsSpaghetti/Pages/Shared/_Layout.cshtml (hunk 5 above)
Comment (discospiff) [reply MomsSpaghetti/Pages/Shared/_Layout.cshtml:16]: Best to use css styling for fonts.

- h3 should be used to describe content, more so than font.
- h3 tag shouldn't be nested inside of an anchor tag.

**Comment by discospiff** [comment]
From this review, I'd harvest the field validation.

For the toString, I recommend the code reviewer explain this to the group first, then let the group make a call on merge.
Fonts: use CSS in the cases presented here, instead of nesting h3 instead of anchor.

File: MomsSpaghetti/Pages/Index.cshtml (hunk 10 above)
Comment (Mohini-Chakravarty) [reply MomsSpaghetti/Pages/Index.cshtml:231]: It has not been added by me. It was already there in the code.

File: MomsSpaghetti/Pages/Index.cshtml (hunk 9 above)
Comment (Mohini-Chakravarty) [reply MomsSpaghetti/Pages/Index.cshtml:234]: It is part of the code that I commented out from the original code. 2,g and pizza are hardcoded values inside the loop where the ingredients had to be displayed. I replaced this part with ingredient.ToString() so as to display the ingredients for the specific recipe.

Comment (Mohini-Chakravarty) [reply MomsSpaghetti/Pages/Index.cshtml:234]: I forgot to remove the commented out part

File: MomsSpaghetti/Pages/Index.cshtml (hunk 11 above)
Comment (Mohini-Chakravarty) [reply MomsSpaghetti/Pages/Index.cshtml:235]: This is all commented out. Part of the previously hard code in the original solution

File: MomsSpaghetti/Pages/Index.cshtml (hunk 1 above)
Comment (Mohini-Chakravarty) [reply MomsSpaghetti/Pages/Index.cshtml:237]: This is all part of the code that has been commented out.

**Comment by TomThomasPaul** [comment]
Implemented required changes. Closing the pull request.

**Pull request closed without merging**